app/employee-app/backend/service.bal

```diff
@@ -0,0 +1,1234 @@
+// Copyright (c) 2024 WSO2 LLC. (http://www.wso2.org).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+import employee_app.authorization;
+import employee_app.database;
+import employee_app.email;
+import employee_app.types;
+
+import ballerina/cache;
+import ballerina/http;
+import ballerina/log;
+import ballerina/time;
+
+configurable email:EmailAlertConfig offerEmailConfig = ?;
+
+final cache:Cache userInfoCache = new ('capacity = types:CAPACITY, 'evictionFactor = types:EVICTION_FACTOR);
+final cache:Cache orgStructureCache = new ('capacity = types:CAPACITY, 'evictionFactor = types:EVICTION_FACTOR);
+
+@display {
+    label: "Employee Application",
+    id: "people-ops/employee-application"
+}
+service http:InterceptableService / on new http:Listener(9090) {
+
+    # Request interceptor.
+    # + return - authorization:JwtInterceptor
+    public function createInterceptors() returns http:Interceptor[] {
+        return [new authorization:JwtInterceptor()];
+    }
+
+    # Get basic information of a given active employee.
+    #
+    # + email - Email of the employee
+    # + return - Basic information of the employee or an error
+    resource function get employees/[string email]()
+        returns types:Employee|http:BadRequest|http:InternalServerError|http:NotFound {
+
+        if !email.matches(types:WSO2_EMAIL_PATTERN) {
+            return <http:BadRequest>{
+                body: {
+                    message: string `Input email is not a valid WSO2 email address: ${email}`
+                }
+            };
+        }
+        types:Employee|error? employee;
+        if userInfoCache.hasKey(email) {
+            any|error cachedEmployee = userInfoCache.get(email);
+            if cachedEmployee is types:Employee {
+                return cachedEmployee;
+            } else if cachedEmployee is cache:Error {
+                log:printError("Error getting employee information", cachedEmployee);
+            }
+        }
+
+        employee = database:getEmployee(email);
+        if employee is error {
+            string customError = string `Error getting employee basic information for ${email}!`;
```

Comment:
Can you use better names for the variables?
```suggestion
            string errorMsg = string `Error getting employee basic information for ${email}!`;
```

Reply:
@sacheeramesh

Reply:
Decided to go with this name after internal discussion.